test/test_find_studies.py

```diff
@@ -0,0 +1,41 @@
+import xml.etree.ElementTree as ET
+
+import pytest
+
+from brdnet import find_studies
+
+
+group1 = {'samples': ['a']}
+group2 = {'samples': ['b']}
+group3 = [{'samples': ['c', 'd']}, {'samples': 'e'}]
+
+study1_in = {'sampleGroups': [group1]}
+study1_expected = ['a']
+study2_in = {'sampleGroups': [{'samples': []}]}
+study2_expected = []
+study3_in = {'sampleGroups': group3}
+study3_expected = ['c', 'd', 'e']
+@pytest.mark.parametrize('test_input,expected', [(study1_in, study1_expected),
+                                                 (study2_in, study2_expected),
+                                                 (study3_in, study3_expected),
+                                                ])
+def test_merge_study_groups(test_input, expected):
+    merged_input = find_studies.merge_study_groups(test_input)
+    assert merged_input == expected
```

Comment:
Seems to me like you only care about the elements in the list and not their order, so maybe you could use something like [unittest assertCountEqual](https://docs.python.org/3.5/library/unittest.html#unittest.TestCase.assertCountEqual) here?

(This is a total nitpick and doesn't really matter for your current test cases, but if you add bigger/more complex test cases it might be good to keep in mind)

Reply:
Oh cool, I hadn't heard of that function before, I'll update it